core/executor/src/wasm_executor.rs

```diff
@@ -427,31 +516,37 @@ impl_function_executor!(this: FunctionExecutor<'e, E>,
 	},
 	// return u32::max_value() if no value exists for the key.
 	ext_get_child_storage_into(
-		storage_key_data: *const u8,
-		storage_key_len: u32,
+		keyspace_data: *const u8,
+		keyspace_len: u32,
+		root_data: *const u8,
+		root_len: u32,
 		key_data: *const u8,
 		key_len: u32,
 		value_data: *mut u8,
 		value_len: u32,
 		value_offset: u32
 	) -> u32 => {
-		let storage_key = this.memory.get(
-			storage_key_data,
-			storage_key_len as usize
-		).map_err(|_| "Invalid attempt to determine storage_key in ext_get_child_storage_into")?;
-		let key = this.memory.get(
-			key_data,
-			key_len as usize
-		).map_err(|_| "Invalid attempt to get key in ext_get_child_storage_into")?;
-
-		let maybe_value = {
-			let storage_key = ChildStorageKey::from_slice(&*storage_key)
-				.ok_or_else(|| "ext_get_child_storage_into: child storage key is not valid")?;
-			this.ext.child_storage(storage_key, &key)
+		let keyspace = &this.memory.get(
+			keyspace_data,
+			keyspace_len as usize,
+		).map_err(|_| "Invalid attempt to determine storage_key in ext_get_allocated_child_storage")?;
+		let key = this.memory.get(key_data, key_len as usize)
+			.map_err(|_| "Invalid attempt to determine key in ext_get_allocated_child_storage")?;
+		let root;
```

Comment:
nit: nop line

Reply:
It is for lifetime resolution, I think there is more elegant way to do it but I do not remember (not the everyday usecase), maybe using a diferent variable name could help.

Reply:
Oh, I see. The first `root` on this line is a `Vec<u8>` and the one on the next line is a `&[u8]` to the first, shadowed variable. I feel like there's a clearer way to express this, but it's not a big deal.